std/traits.d

```diff
@@ -4543,6 +4543,41 @@ unittest
     static assert(!__traits(compiles, lvalueOf!byte = 128));
 }
 
+/+
+Given a type $(D T), $(D lvalueInit) represents a unique instance of $(D T)
+initialized to $(D T.init). $(D lvalueInit) is always immutable, regardless
+of $(D T)'s qualifiers.
+
+$(D lvalueInit) is useful when one needs a reference the $(D .init) value,
+without creating duplicate static instances, or temporaries that would require
+destruction.
++/
+package template lvalueInit(T)
+{
+    //static if (is(T == immutable(T)))
+    //    static lvalueInit = T.init;
+    //else
+    //    alias lvalueInit = .lvalueInit!(immutable(T));
+    static lvalueInit = immutable(T).init;
```

Comment:
If you want something that is only for template constraints, we already have `lvalueof` in `std.traits` (http://dlang.org/phobos/std_traits.html#lvalueOf). It has the exact signature you propose (give or take an "inout") workaround.

However, I see this as different concepts, since the idea here is to *specifically* be able to grab a value. You say "impossible to misuse " but:
- `lvalueInit` is immutable, so *can't* be written too anyways.
- Using a function (I *also* objected to this with `lvalueof`) means operator & has a surprising effect, in that it doesn't give you the address of your lvalue, which, IMO, kind of breaks the intent of the damn thing.